libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Extensions/RestClient.swift

```diff
@@ -179,4 +266,43 @@ extension RestClient {
             }
         }
     }
+  
+    // MARK: Record Convience API - Swift & Combine
+
+  
+    /// This method provides a reusuable, generic Combine pipeline for retrieving records
+    ///   from Salesforce. It relys on Swift Generics, and type inference to determine what
+    ///  models to create.
+    ///
+    /// Given a model object - Contact, you can use this pipeline like this:
+    /// contactsForCancellable = RestClient.shared.records(forRequest: request)
+    ///   .receive(on: RunLoop.main)
+    ///   .assign(to: \.contacts, on: self)
+    ///
+    /// This pipeline infers it's return type from the variable in the assign subscriber.
+    func records<Record: Decodable>(forRequest request:RestRequest ) -> AnyPublisher<[Record], Never> {
+      guard request.isQueryRequest else {
+        return Empty(completeImmediately: true).eraseToAnyPublisher()
+      }
+      return RestClient.shared.publisher(for: request)
+        .tryMap({ (response) -> Data in
+          response.asData()
+        })
+        .decode(type: QueryResponseWrapper<Record>.self, decoder: JSONDecoder())
+        .map({ (record) -> [Record] in
+          record.records
+        })
+        .catch({ _ in
```

Comment:
@codefriar Similar comment about swallowing the error.

Reply:
This is now returning an empty query response as well

Reply:
thats not true anymore. I just refactored so the pipeline is defined AnyPublisher<QueryResponse<Record>, RestClientError> so now'll throw the error up the stack.